### PR TITLE
extract gyroscope sensor and show delta between true and detected data

### PR DESCRIPTION
**Pull request by AlexPascu001**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -33,42 +40,57 @@ def register_input():
...
 
     # Render the observation using pygame.
     # Rotate + flip the observation, see: 
     # https://stackoverflow.com/questions/66241275/pygame-rotates-camera-stream-from-opencv-camera.
-    rotated = np.rot90(observation, k=1, axes=(0,1))
+    rotated = np.rot90(observation, k=1, axes=(0, 1))
     flipped = np.flip(rotated, axis=0)
     extracted_speed = feature_extraction.extract_true_speed(observation)
     extracted_abs = feature_extraction.extract_abs(observation)
+    indicator_bar = feature_extraction.extract_indicators(flipped)
```
Comment (visanalexandru) [inline main.py]: Why not use the observation?

File: main.py [hunk 2]
```diff
@@ -33,42 +40,57 @@ def register_input():
...
+    current_gyroscope_value = feature_extraction.extract_gyroscope(indicator_bar_resized)
+    current_gyroscope_value = current_gyroscope_value / MAX_GYROSCOPE_VAL
+
+    # print(f"True speed: {true_speed:.2f} | Extracted speed: {extracted_speed:.2f}")
+    # print(f"True abs: {true_abs} | Extracted abs: {extracted_abs}")
+    # print(f"Current gyroscope value: {current_gyroscope_value:.2f}")
+
+    print(f"Delta true speed: {true_speed - extracted_speed:.2f} | Delta true abs: {true_abs[0] - extracted_abs[0]:.2f} | Gyroscope: {current_gyroscope_value:.2f}")
+
+    surface = pygame.surfarray.make_surface(indicator_bar)
```
Comment (visanalexandru) [inline main.py]: Can we please render the game instead?

File: feature_extraction.py [hunk 3]
```diff
@@ -66,9 +70,44 @@ def extract_abs(image):
...
+    upper_red = np.array([255, 80, 80])
+
+    # Create a mask that isolates the red area
+    mask = cv2.inRange(image_rgb, lower_red, upper_red)
+
+    # Apply the mask to get the isolated red area
+    isolated_red = cv2.bitwise_and(image_rgb, image_rgb, mask=mask)
+
+    # Find the contours of the isolated red area
+    contours, _ = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
```
Comment (visanalexandru) [inline feature_extraction.py:100]: Great idea using `cv2.findContours`. I will use it also!

File: main.py (hunk 1 above)
Comment (AlexPascu001) [reply main.py]: Fine, I will change this to use observation

File: main.py (hunk 2 above)
Comment (AlexPascu001) [reply main.py]: Sure, this was a leftover that wasn't supposed to come out in the PR

**Review by visanalexandru — APPROVED** [review]

**Pull request merged**